packages/headless/src/api/service/insight/insight-api-client.ts

```diff
@@ -81,10 +82,12 @@ export class InsightAPIClient {
   }
 
   async query(
-    req: InsightQueryRequest
+    req: InsightQueryRequest,
+    options?: SearchOptions
   ): Promise<InsightAPIResponse<SearchResponseSuccess>> {
     const response = await PlatformClient.call({
       ...buildInsightQueryRequest(req),
+      requestMetadata: {method: 'query', origin: options?.origin},
```

Comment:
```suggestion
      requestMetadata: {method: 'search', origin: options?.origin},
```

Reply:
It's more about the action that will be executed on the Search API. In the insight use case it's a bit special because the requests are not sent directly sent to the search API, they are sent to Customer Service API which performs the request to the search API. so the finality of the request is a search request.
That's why imo the value should be `search`.